File: web-app/online-warehouse/src/app/write-off-act/dto/create-write-off-act-goods.dto.ts
import {WriteOffTypeEnum} from "./enum/write-off-type.enum";

export class CreateWriteOffActGoodsDto {

  goodsId: number;
  writeOffType: WriteOffTypeEnum;
  amount: number;

  constructor(goodsId: number, writeOffType: WriteOffTypeEnum, amount: number) {
    this.goodsId = goodsId;
    this.writeOffType = writeOffType;
    this.amount = amount;
  }
}

File: core/src/main/java/by/itechart/writeoffact/dto/CreateWriteOffActGoodsDto.java
package by.itechart.writeoffact.dto;

import by.itechart.writeoffact.enums.WriteOffType;
import lombok.Data;

import javax.validation.constraints.NotNull;

@Data
public class CreateWriteOffActGoodsDto {
    private Long goodsId;
    @NotNull
    private WriteOffType writeOffType;
    @NotNull
    private Integer amount;
}
